### PR TITLE
llama4 npu adapt

### PR DESCRIPTION
**Pull request by KnightLTC**
## Motivation

llama4 npu adapt

## Modifications

1. Add ascend in attention backend list for llama4

## Accuracy Tests
NPU
image
GPU
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
3. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
4. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
5. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/models/llama4.py [hunk 1]
```diff
@@ -116,6 +118,7 @@ def __init__(
             hidden_size=config.hidden_size,
             intermediate_size=intermediate_size_moe,
             layer_id=layer_id,
+            top_k=self.top_k,
```
Comment (xiaobaicxy) [inline python/sglang/srt/models/llama4.py]: Please explain why FusedMoE requires top-k on NPU, but not on GPU. Please add the GPU test results in the PR description.

File: python/sglang/srt/models/llama4.py [hunk 2]
```diff
@@ -328,6 +331,8 @@ def forward(
         if self.rotary_emb is not None:
             q_view, k_view = qk.split([self.q_size, self.kv_size], dim=-1)
             q_out_unused, k_out_unused = self.rotary_emb(positions, q_view, k_view)
+            if _is_npu:
+                qk = torch.cat([q_out_unused, k_out_unused], dim=-1)
```
Comment (xiaobaicxy) [inline python/sglang/srt/models/llama4.py:335]: Please add a comment explaining why the NPU requires q_out_unused and k_out_unused, while the GPU does not.

File: python/sglang/srt/models/llama4.py (hunk 1 above)
Comment (KnightLTC) [reply python/sglang/srt/models/llama4.py]: When top_k is none, num_token * top_k will report an None error, so top_k needs to be passed

File: python/sglang/srt/models/llama4.py (hunk 2 above)
Comment (KnightLTC) [reply python/sglang/srt/models/llama4.py:335]: Llama4 has accuracy issues on NPU and needs to be added to improve accuracy

File: python/sglang/srt/models/llama4.py [hunk 3]
```diff
@@ -116,6 +118,9 @@ def __init__(
             hidden_size=config.hidden_size,
             intermediate_size=intermediate_size_moe,
             layer_id=layer_id,
+            top_k=(
+                None if not _is_npu else self.top_k
+            ),  # The top_k parameter is used in NPU scenarios
```
Comment (iforgetmyname) [inline python/sglang/srt/models/llama4.py]: why on npu we need this change? is there any error reported when not passing this topk?

Comment (KnightLTC) [reply python/sglang/srt/models/llama4.py]: yes. Modifications has been updated

Comment (Hexq0210) [inline python/sglang/srt/models/llama4.py]: self.topk is TopK() object now,we need config.num_experts_per_tok

Comment (KnightLTC) [reply python/sglang/srt/models/llama4.py]: it is int
